public/static/js/projects.js

```diff
@@ -431,21 +423,25 @@ mciModule.controller('ProjectCtrl', function($scope, $window, $http, $location,
   };
 
   $scope.addGithubAlias = function() {
-    if ($scope.github_alias.variant && $scope.github_alias.task) {
-      item = Object.assign({}, $scope.github_alias)
-      item["alias"] = "__github"
-      $scope.github_aliases = $scope.github_aliases.concat([item]);
-      delete $scope.github_alias
-      $scope.invalidPatchDefinitionMessage = "";
-    }
+    if (!$scope.validPatchDefinition($scope.github_alias)) {
+      $scope.invalidPatchDefinitionMessage = "Patch should has variant and task xor tags"
+      return
+    }
+    item = Object.assign({}, $scope.github_alias)
+    item["alias"] = "__github"
+    $scope.github_aliases = $scope.github_aliases.concat([item]);
+    delete $scope.github_alias
+    $scope.invalidPatchDefinitionMessage = "";
   };
 
   $scope.addPatchAlias = function() {
-    if ($scope.patch_alias.alias && $scope.patch_alias.variant && ($scope.patch_alias.task || $scope.patch_alias.tags_temp)) {
-      item = Object.assign({}, $scope.patch_alias)
-      $scope.patch_aliases = $scope.patch_aliases.concat([item]);
-      delete $scope.patch_alias
+    if (!$scope.validPatchAlias($scope.patch_alias)){
+      $scope.invalidPatchAliasMessage = "Patch should has alias, variant and task xor tags"
```

Comment:
"A patch alias must have an alias name, variant regex, and exactly one of task regex or tag"